app/experimenter/static/rapid/components/forms/ExperimentForm.tsx

```diff
@@ -22,26 +28,32 @@ const ErrorList: React.FC<ErrorListProperties> = ({ errors }) => {
 };
 
 const ExperimentForm: React.FC = () => {
-  const [formData, setFormData] = React.useState({
-    name: "",
-    objectives: "",
-  });
+  const formData = useExperimentState();
+  const dispatch = useExperimentDispatch();
+  const { experimentSlug } = useParams();
+
   const [errors, setErrors] = React.useState<{ [name: string]: Array<string> }>(
     {},
   );
   const history = useHistory();
 
   const handleChange = (ev) => {
     const field = ev.target;
-    setFormData({
-      ...formData,
-      [field.getAttribute("name")]: field.value,
+    dispatch({
+      type: ExperimentReducerActionType.UPDATE_STATE,
+      state: {
+        ...formData,
+        [field.getAttribute("name")]: field.value,
+      },
     });
   };
 
   const handleClickSave = async () => {
-    const response = await fetch("/api/v3/experiments/", {
-      method: "POST",
+    const url = experimentSlug
```

Comment:
I noticed that if the whole request fails (like right now, the`PUT` request doesn't work) we don't have a general error to display – can you file an issue for that? (also maybe to implement editing)

Reply:
the `PUT` request should work?? we should handle errors better but curious how you got it to fail?

Reply:
I just got a `400` response

Reply:
Oh I'm getting that too now I guess when I rebased the form become incompatible because it's missing features.

Reply:
It's not showing the error message because the field doesn't match the updated API, we were looking for `feature` and it should be `features`. I've updated that locally and it should be fixed.

Reply:
ah got it. yeah you;re right there is a features error message